### PR TITLE
Transformer CDK: solving cdk new destination onboarding problem

### PR DESCRIPTION
**Pull request by shrouti1507**
## Description of the change

Previously, only old destinations were being onboarded to CDK. While onboarding new destinations the file path was not matching with new cdk format.

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: versionedRouter.js [hunk 1]
```diff
@@ -90,8 +91,12 @@ const functionsEnabled = () => {
 };
 
 async function handleDest(ctx, version, destination) {
-  const destHandler = getDestHandler(version, destination);
+  // const destHandler = getDestHandler(version, destination);
+  let destHandler;
   const events = ctx.request.body;
+  if (!events || !Array.isArray(events)) {
+    throw new Error("Event is missing or in inappropriate format", 400);
```
Comment (sanpj2292) [inline versionedRouter.js:98]: `Error` class in Javascript doesn't have 2nd argument AFAIK, please change this to `CustomError`

Although this doesn't throw a runtime error because we used `Error` here but since it is an error thrown on transformer, the status code returned will be `5xx` only through the route to server, so it makes the 2nd argument(where `400` has been set) useless

Comment (sanpj2292) [inline versionedRouter.js:98]: `Error` class in Javascript doesn't have 2nd argument AFAIK, please change this to `CustomError`

Although this doesn't throw a runtime error because we used `Error` here but since it is an error thrown on transformer, the status code returned will be `5xx` only through the route to server, so it makes the 2nd argument(where `400` has been set) useless

File: versionedRouter.js [hunk 2]
```diff
@@ -177,7 +186,6 @@ async function handleDest(ctx, version, destination) {
   ctx.body = respList;
   return ctx.body;
 }
-
```
Comment (saikumarrs) [inline versionedRouter.js:180]: Revert this change.

File: versionedRouter.js [hunk 3]
```diff
@@ -106,6 +111,10 @@ async function handleDest(ctx, version, destination) {
   });
   const respList = [];
   const executeStartTime = new Date();
+  // destination definition is going to be same for any kind of event
+  if (!isCdkDestination(events[0])) {
+    destHandler = getDestHandler(version, destination);
+  }
```
Comment (saikumarrs) [inline versionedRouter.js:117]: There is a bug on Ln 127. It should have been `parsedEvent` rather than `event`.
CC @sanpj2292

File: versionedRouter.js [hunk 4]
```diff
@@ -90,8 +91,12 @@ const functionsEnabled = () => {
 };
 
 async function handleDest(ctx, version, destination) {
-  const destHandler = getDestHandler(version, destination);
+  // const destHandler = getDestHandler(version, destination);
+  let destHandler;
```
Comment (saikumarrs) [inline versionedRouter.js:95]: You can relocate this declaration after Ln 113.

File: versionedRouter.js [hunk 5]
```diff
@@ -90,8 +91,12 @@ const functionsEnabled = () => {
 };
 
 async function handleDest(ctx, version, destination) {
-  const destHandler = getDestHandler(version, destination);
+  // const destHandler = getDestHandler(version, destination);
```
Comment (saikumarrs) [inline versionedRouter.js:94]: Remove this.

File: versionedRouter.js [hunk 6]
```diff
@@ -90,8 +91,12 @@ const functionsEnabled = () => {
...
+  let destHandler;
   const events = ctx.request.body;
+  if (!events || !Array.isArray(events)) {
```
Comment (saikumarrs) [inline versionedRouter.js:97]: Also, check for the length of the array.

Comment (saikumarrs) [inline versionedRouter.js:97]: You can skip `!events` check as `Array.isArray` takes care of that.

File: versionedRouter.js (hunk 3 above)
Comment (sanpj2292) [reply versionedRouter.js:117]: Changed it, can you check now ?

File: versionedRouter.js [hunk 7]
```diff
@@ -90,8 +92,10 @@ const functionsEnabled = () => {
 };
 
 async function handleDest(ctx, version, destination) {
-  const destHandler = getDestHandler(version, destination);
   const events = ctx.request.body;
+  if (!Array.isArray(events) && events.length === 0) {
```
Comment (saikumarrs) [inline versionedRouter.js:96]:
```suggestion
  if (!Array.isArray(events) || events.length === 0) {
```

**Review by saikumarrs — DISMISSED** [review]

File: versionedRouter.js [hunk 8]
```diff
@@ -106,16 +110,21 @@ async function handleDest(ctx, version, destination) {
   });
   const respList = [];
   const executeStartTime = new Date();
+  let destHandler;
+  // destination definition is going to be same for any kind of event
```
Comment (utsabc) [inline versionedRouter.js:114]: Some inline documentation for this would make things easier to understand

Comment (sanpj2292) [reply versionedRouter.js]: Can you now check if this is making it easier to understand ?

**Review by saikumarrs — APPROVED** [review]

**Review by utsabc — APPROVED** [review]
